src/content/en/tools/workbox/guides/advanced-recipes.md

```diff
@@ -15,32 +15,42 @@ worker has updated and waiting to install.
 
 To do this you'll need to add some code to your page and to your service worker.
 
-**Add to your page**
+**Code in your page**
 
 ```html
 <script type="module">
-import {Workbox} from 'https://storage.googleapis.com/workbox-cdn/releases/4.0.0/workbox-window.prod.mjs';
+import {Workbox} from 'https://storage.googleapis.com/workbox-cdn/releases/4.1.0/workbox-window.prod.mjs';
 
 if ('serviceWorker' in navigator) {
   const wb = new Workbox('/sw.js');
 
   // Add an event listener to detect when the registered
   // service worker has installed but is waiting to activate.
-  wb.addEventListener('waiting', () => {
+  wb.addEventListener('waiting', (event) => {
+    // `event.wasWaitingBeforeRegister` will be true if this is
+    // the first time the updated service worker is waiting.
+    // When `event.wasWaitingBeforeRegister` is false, a previously
+    // updated same service worker is still waiting.
+    // You may want to customize the UI prompt accordingly.
+
     // Assumes your app has some sort of prompt UI element
     // that a user can either accept or reject.
     const prompt = createUIPrompt({
       onAccept: async () => {
-        // Send a message to the service worker telling it to
-        // skip waiting. If/when the service worker replies,
-        // you know that `skipWaiting()` was called.
+        // Assuming the user accepted the update, set up a listener
+        // that will reload the page as soon as the previously waiting
+        // service worker has taken control.
+        wb.addEventListener('controlling', (event) => {
+          window.location.reload();
```

Comment:
I don't think we need to wait for controlling here, do we?

Reply:
I think it's safest to do that, since it's the only way to be sure that the new SW will control the page after the reload is if the controller has actually changed.